shin-core/src/format/scenario/info.rs

```diff
@@ -80,34 +115,220 @@ impl MovieInfoItem {
+#[derive(Debug, BinRead, BinWrite)]
+pub enum CharsGridSegment {
+    /// Ends the current character grid definition.
+    #[brw(magic = 0x0u8)]
+    End,
+
+    /// Defines a portrait on the grid, with a corresponding character to have its full sprite, name, and description shown when the portrait is selected.
+    #[brw(magic = 0x1u8)]
+    Portrait {
+        /// The page to show this portrait on, between 0 and 3 (both inclusive). The game will automatically generate as many pages as necessary for a specific grid.
+        page: u8,
+
+        /// The portrait's X position on the grid.
+        grid_x: u8,
+
+        /// The portrait's Y position on the grid.
+        grid_y: u8,
+
+        /// The ID of the character this portrait is for, indexing into [`CharsSpriteInfo`].
+        character_id: u16,
+
+        /// Defines how the portrait and the corresponding sprite is displayed, and how it behaves on Execute/Resurrect:
+        ///  - **in game**: `0` = empty portrait, `1` = alive, `2` = dead (small blood splotch on sprite), `3` = very dead (large blood splotch on sprite)
+        ///  - **out of game** (Characters screen selected from main menu): `0` = empty portrait, `1` = two phases (alive/dead), `2` = three phases (alive/missing/dead)
+        behaviour: u8,
+
+        /// Modifies the `behaviour` in certain circumstances.
+        behaviour_modifier: u8, // todo: find out and document what this does
+    },
+
+    /// Defines an individual line segment to be placed on the grid, to connect character portraits.
+    #[brw(magic = 0x2u8)]
+    Connector {
+        /// The page to show this portrait on, between 0 and 3 (both inclusive). The game will automatically generate as many pages as necessary for a specific grid.
+        page: u8,
+
+        /// The connector's X position on the grid.
+        grid_x: u8,
+
+        /// The connector's Y position on the grid.
+        grid_y: u8,
+
+        /// The shape of this connector:
+        ///  - `0` = none
+        ///  - `1` = left dead end
+        ///  - `2` = top dead end
+        ///  - `3` = left-top elbow
+        ///  - `4` = right dead end
+        ///  - `5` = horizontal line
+        ///  - `6` = top-right elbow
+        ///  - `7` = T to the top
+        ///  - `8` = bottom dead end
+        ///  - `9` = bottom-left elbow
+        ///  - `10` = vertical line
+        ///  - `11` = T to the left
+        ///  - `12` = right-bottom elbow
+        ///  - `13` = T to the bottom
+        ///  - `14` = T to the right
+        ///  - `15` = cross
+        ///  - `16` = dark vertical line
+        shape: u8,
+
+        /// The color of this connector: `1` = red, `2` = blue, `3` = yellow.
+        color: u8,
+    },
+}
+
+impl FinalSegment for CharsGridSegment {
+    fn is_final(&self) -> bool {
+        match self {
+            CharsGridSegment::End => true,
+            _ => false,
+        }
+    }
+}
+
+#[derive(Debug, BinRead, BinWrite)]
+pub struct CharsGridInfoItem {
```

Comment:
What do the item ids correspond to? Episodes?

Reply:
IDs 0 to 7 correspond to the final grids for episode 1-8 that can be accessed from the title screen. The other IDs (there are like 150) correspond to individual grids representing the state of the characters at specific points of an episode. The currently active grid is selected using the `CHARS` (`0xe0`) instruction

Reply:
Now I get it! thx